cmd/controller-manager/main.go

```diff
@@ -189,62 +190,74 @@ func main() {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
-
 }
 
-func watchConfig(configWatcher *watcher.K8sConfigMapWatcher, cfg *config.Config, stopCh <-chan struct{}) {
-	go func() {
-		// watch for reconciliation signals, and grab configmaps, then update the running configuration
-		// for the server
-		sigChan := make(chan interface{}, 10)
-		//debouncedChan := make(chan interface{}, 10)
+func setupWatchQueue(stopCh <-chan struct{}, configWatcher *watcher.K8sConfigMapWatcher) workqueue.Interface {
+	// watch for reconciliation signals, and grab configmaps, then update the running configuration
+	// for the server
+	sigChan := make(chan interface{}, 10)
 
-		// debounce events from sigChan, so we dont hammer apiserver on reconciliation
-		eventsCh := utils.Coalescer(EventCoalesceWindow, sigChan, stopCh)
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
```

Comment:
I think it might be too frequent than the previous one.

How about reconsidering customizing the `RateLimiter`?

And I think we don't need `EventCoalesceWindow`  anymore, please delete it.

Reply:
OK! I set qps to 0.5